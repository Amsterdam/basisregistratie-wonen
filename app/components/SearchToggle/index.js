import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import SearchIcon from '@datapunt/asc-assets/lib/Icons/Search.svg';
import CloseIcon from '@datapunt/asc-assets/lib/Icons/Close.svg';

const StyledIcon = styled.button`
  padding: 10px;
  z-index: 1;
  width: 50px;
  height: 50px;
  border: 0;
  background: none;

  &:hover {
    background-color: #ec0000;
    cursor: pointer;

    svg {
      fill: white;
    }
  }

  &:focus {
    outline: 4px solid #fec813;
  }
`;

const SearchToggle = ({ active, onClick, className, label, ...rest }) => (
  <StyledIcon size={20} padding={0} className={className} onClick={onClick} type="button" {...rest}>
    {active ? (
      <CloseIcon data-testid="searchToggle-close-icon" focusable="false" width={20} />
    ) : (
      <SearchIcon data-testid="searchToggle-search-icon" focusable="false" width={20} />
    )}
    <span className="visuallyhidden" data-testid="searchToggle-label">
      {label}
    </span>
  </StyledIcon>
);

SearchToggle.defaultProps = {
  active: false,
  className: '',
  onClick: null,
};

SearchToggle.propTypes = {
  active: PropTypes.bool,
  className: PropTypes.string,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};

export default SearchToggle;
