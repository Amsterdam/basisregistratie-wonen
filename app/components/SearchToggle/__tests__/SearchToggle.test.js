import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import 'jest-styled-components';
import SearchToggle from '..';

describe('SearchToggle', () => {
  afterEach(cleanup);

  it('matches the snapshot', () => {
    const { container, rerender, queryByTestId } = render(<SearchToggle label="Foo bar" />);

    expect(queryByTestId('searchToggle-close-icon')).toBeNull();
    expect(queryByTestId('searchToggle-search-icon')).not.toBeNull();
    expect(queryByTestId('searchToggle-label').classList).toContain('visuallyhidden');
    expect(container.firstChild).toMatchSnapshot();

    rerender(<SearchToggle label="Foo bar" active />);

    expect(queryByTestId('searchToggle-close-icon')).not.toBeNull();
    expect(queryByTestId('searchToggle-search-icon')).toBeNull();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('calls the onClick handler', () => {
    const onClick = jest.fn();
    const { container } = render(<SearchToggle label="Foo bar" onClick={onClick} />);

    fireEvent(
      container.firstChild,
      new MouseEvent('click', {
        bubbles: true,
      }),
    );

    expect(onClick).toHaveBeenCalled();
  });
});
