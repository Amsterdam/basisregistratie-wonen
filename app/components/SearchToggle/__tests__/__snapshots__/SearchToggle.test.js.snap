// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchToggle matches the snapshot 1`] = `
.c0 {
  padding: 10px;
  z-index: 1;
  width: 50px;
  height: 50px;
  border: 0;
  background: none;
}

.c0:hover {
  background-color: #ec0000;
  cursor: pointer;
}

.c0:hover svg {
  fill: white;
}

.c0:focus {
  outline: 4px solid #fec813;
}

<button
  class="c0"
  size="20"
  type="button"
>
  <img
    data-testid="searchToggle-search-icon"
    width="20"
  />
  <span
    class="visuallyhidden"
    data-testid="searchToggle-label"
  >
    Foo bar
  </span>
</button>
`;

exports[`SearchToggle matches the snapshot 2`] = `
<button
  class="SearchToggle__StyledIcon-qg6jkp-0 bcSxwk"
  size="20"
  type="button"
>
  <img
    data-testid="searchToggle-close-icon"
    width="20"
  />
  <span
    class="visuallyhidden"
    data-testid="searchToggle-label"
  >
    Foo bar
  </span>
</button>
`;
